### PR TITLE
improvement: improve error handling and logging

### PR DESCRIPTION
**Pull request by mateo-ivc**
## Description

Improve and unify error handling and logging
## Changelog

- In almost all /api/... files added missing error handling or logging
- Unified error handling and logging
## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] The light- and dark-theme are both supported and tested
- [ ] The design was implemented and is responsive for all devices and screen sizes
- [x] The application was tested in the most commonly used browsers (e.g. Chrome, Firefox, Safari)

### CODE REVIEW

File: server/src/api/login.go [hunk 1]
```diff
@@ -28,21 +28,21 @@ func (s *Server) signInAnonymously(w http.ResponseWriter, r *http.Request) {
...
-		common.Throw(w, r, common.InternalServerError)
+		common.Throw(w, r, err)
 		return
 	}
 
 	tokenString, err := s.auth.Sign(map[string]interface{}{"id": user.ID})
 	if err != nil {
 		log.Errorw("unable to generate token string", "err", err)
-		common.Throw(w, r, common.InternalServerError)
+		common.Throw(w, r, err)
```
Comment (Schwehn42) [inline server/src/api/login.go:45]: why aren't you throwing interval server error explicitly anymore?

Comment (mateo-ivc) [reply server/src/api/login.go:45]: I think the plan was to return a more meaningful error.
But common.Throw() also just renders an InternalServerError when not specifying an ApiError.
Either way nothing really changed and my changes were meaningless :D
I'll change it back :)

Comment (Schwehn42) [reply server/src/api/login.go:45]: no it's fine then 👍  
Or alternatively we could add more custom errors to get more variety than 401, 404 and 500?

**Comment by mateo-ivc** [comment]
outdated

**Pull request closed without merging**